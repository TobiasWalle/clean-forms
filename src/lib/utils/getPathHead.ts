import { Path } from './FieldRegister';
import { transformPathToArray } from './transformPathToArray';

export function getPathHead(path: Path): string {
  const pathAsArray = transformPathToArray(path);
  const isEmpty = pathAsArray.length === 0;
  if (isEmpty) {
    throw new ReferenceError('Path is empty');
  }
  const lastIndex = pathAsArray.length - 1;
  return pathAsArray[lastIndex];
}

export function getPathHeadAsNumber(path: Path): number {
  const pathHead = getPathHead(path);
  const pathHeadAsNumber = Number(pathHead);
  if (isNaN(pathHeadAsNumber)) {
    throw new Error(`Cannot convert the path head "${pathHead}" to a number`);
  }
  return pathHeadAsNumber;
}
