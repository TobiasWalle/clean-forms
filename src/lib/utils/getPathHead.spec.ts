import { getPathHead, getPathHeadAsNumber } from './getPathHead';

describe('getPathHead', () => {
  it('should get path head', () => {
    expect(getPathHead('a.b')).toBe('b');
  });

  it('should throw error if path is empty', () => {
    expect(() => getPathHead('')).toThrowErrorMatchingSnapshot();
  });
});

describe('getPathHeadAsNumber', () => {
  it('should get path head as number', () => {
    expect(getPathHeadAsNumber('a.0')).toBe(0);
  });

  it('should throw error if path cannot be a number', () => {
    expect(() => getPathHeadAsNumber('a')).toThrowErrorMatchingSnapshot();
  });
});
