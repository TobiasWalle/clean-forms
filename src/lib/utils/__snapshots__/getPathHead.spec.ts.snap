// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPathHead should throw error if path is empty 1`] = `"Path is empty"`;

exports[`getPathHeadAsNumber should throw error if path cannot be a number 1`] = `"Cannot convert the path head \\"a\\" to a number"`;
